### PR TITLE
Fix misc inconsistencies between code and doc comments

### PR DESCRIPTION
**Pull request by TysonAndre**
Various issues were detected by a static analyzer

### CODE REVIEW

File: system/core/Controller.php [hunk 1]
```diff
@@ -48,6 +48,19 @@
  * @category	Libraries
  * @author		EllisLab Dev Team
  * @link		https://codeigniter.com/user_guide/general/controllers.html
+ *
+ * Note: the below aren't magic properties, they are dynamic properties (Set in the constructor, based on calls to load_class)
+ *
+ * @property CI_Benchmark	$benchmark (core)
```
Comment (narfbg) [inline system/core/Controller.php]: We don't use `@property`.

File: system/core/Exceptions.php [hunk 2]
```diff
@@ -239,7 +239,7 @@ public function show_exception($exception)
 	 * @param	string	$message	Error message
 	 * @param	string	$filepath	File path
 	 * @param	int	$line		Line number
-	 * @return	string	Error page output
+	 * @return	void	Echoes the error page output to stdout
```
Comment (narfbg) [inline system/core/Exceptions.php]: `void` is correct, indeed.
But the description has nothing to do with the return type.

File: system/core/compat/mbstring.php [hunk 3]
```diff
@@ -97,7 +97,7 @@ function mb_strlen($str, $encoding = NULL)
 	 * @param	string	$needle
 	 * @param	int	$offset
 	 * @param	string	$encoding
-	 * @return	mixed
+	 * @return	string|false
```
Comment (narfbg) [inline system/core/compat/mbstring.php]: We don't use `this|that` ... even if we did, `bool` would be the correct thing.
Either way, `false` is a rare edge case that we don't care for.

Comment (narfbg) [inline system/core/compat/mbstring.php]: On another note ... `string` is also incorrect. It returns an integer.
Better leave it `mixed`.

File: system/database/DB_forge.php [hunk 4]
```diff
@@ -488,7 +488,7 @@ public function drop_table($table_name, $if_exists = FALSE)
 	 *
 	 * @param	string	$table		Table name
 	 * @param	bool	$if_exists	Whether to add an IF EXISTS condition
-	 * @return	string
+	 * @return	string|TRUE (Returns TRUE to indicate there's nothing to do)
```
Comment (narfbg) [inline system/database/DB_forge.php]: Same as `mb_strpos()` (on the `|true` thing).

File: system/database/DB_forge.php [hunk 5]
```diff
@@ -992,7 +992,7 @@ protected function _process_primary_keys($table)
 	 * Process indexes
 	 *
 	 * @param	string	$table
-	 * @return	string
+	 * @return	string[] list of SQL statements
```
Comment (narfbg) [inline system/database/DB_forge.php]: ... it's an internal function. Why do you care?

Also, if you're fixing "inconsistencies", why does the return has a description, but the parameter not?

**Comment by narfbg** [comment]
What is "the master branch of phan" and why should we care for it?

File: system/core/Controller.php (hunk 1 above)
Comment (TysonAndre) [inline system/core/Controller.php]: removed.

File: system/core/Exceptions.php (hunk 2 above)
Comment (TysonAndre) [inline system/core/Exceptions.php]: Moved to the top of the phpdoc description.

File: system/core/compat/mbstring.php (hunk 3 above)
Comment (TysonAndre) [inline system/core/compat/mbstring.php]: Right, should be int, not string. Changed back to mixed

File: system/database/DB_forge.php (hunk 4 above)
Comment (TysonAndre) [inline system/database/DB_forge.php]: Changed to mixed

File: system/database/DB_forge.php (hunk 5 above)
Comment (TysonAndre) [inline system/database/DB_forge.php]: I meant type inconsistencies.

**Comment by TysonAndre** [comment]
> What is "the master branch of phan" and why should we care for it?

Removed mention of phan from the commit description.

https://github.com/etsy/phan/ is a static analyzer for PHP that prefers to minimize false-positives.

File: system/core/Exceptions.php [hunk 6]
```diff
@@ -234,12 +234,13 @@ public function show_exception($exception)
 
 	/**
 	 * Native PHP error handler
+     * (Echoes the error page output to stdout)
```
Comment (narfbg) [inline system/core/Exceptions.php]: Uses spaces for indentation (we use tabs) ... diff itself makes it look obvious. :)

Does this really need to be said in a docblock? The function name itself describes it pretty well.

File: system/core/Security.php [hunk 7]
```diff
@@ -594,7 +594,7 @@ public function xss_hash()
 	 * Get random bytes
 	 *
 	 * @param	int	$length	Output length
-	 * @return	string
+	 * @return	string|false
```
Comment (narfbg) [inline system/core/Security.php]: `false` is an edge case that could never happen in a sane environment. We don't care for it.

Comment (TysonAndre) [inline system/core/Security.php]: Made the requested changes to Security and Exceptions

File: system/core/Exceptions.php (hunk 6 above)
Comment (TysonAndre) [inline system/core/Exceptions.php]: Removed that line and the spaces

**Comment by narfbg** [comment]
>> What is "the master branch of phan" and why should we care for it?
>
> Removed mention of phan from the commit description.
>
> https://github.com/etsy/phan/ is a static analyzer for PHP that prefers to minimize false-positives.

I see ...

Well, good if it helps you, but I absolutely despise the idea of some third-party tool dictating changes to our code. Please only change things that matter to you.

**Pull request merged**